src/main/java/com/cloudant/client/org/lightcouch/CouchDbClientBase.java

```diff
@@ -307,20 +241,29 @@ public Replicator replicator() {
 
     /**
      * Executes a HTTP request.
-     * <p><b>Note</b>: The response must be closed after use to release the connection.
+     * <p><b>Note</b>: The stream must be closed after use to release the connection.
      *
-     * @param request The HTTP request to execute.
-     * @return {@link HttpResponse}
+     * @param connection The HTTP request to execute.
+     * @return Class type of object T (i.e. {@link Response}
      */
-    public HttpResponse executeRequest(HttpRequestBase request) {
+    public <T> T executeRequest(HttpConnection connection, Class<T> classType) {

```

Comment:
Yes - currently the only return types are InputStream and Response.  Maybe it's better to just have two methods?